From f215e451a9eb799140487615c02ffae4a680ab85 Mon Sep 17 00:00:00 2001
From: Shengjiu Wang <shengjiu.wang@freescale.com>
Date: Thu, 16 Feb 2017 11:51:06 +0800
Subject: [PATCH 1519/5242] MLK-13980: ASoC: fsl_rpmsg_i2s: fix wrong cmd sent
 to M4 in suspend

commit  6620cb3d44af5a65476876fbd68b71a8c9cb113d from
https://source.codeaurora.org/external/imx/linux-imx.git

This typo issue will cause that wrong cmd send to M4 side.

Fixes: 3e13a631aee0 ("MLK-13904-1: ASoC: fsl: add audio cpu dai driver base on rpmsg")

Signed-off-by: Shengjiu Wang <shengjiu.wang@freescale.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 sound/soc/fsl/fsl_rpmsg_i2s.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/sound/soc/fsl/fsl_rpmsg_i2s.c b/sound/soc/fsl/fsl_rpmsg_i2s.c
index 2768c65..c53807f 100644
--- a/sound/soc/fsl/fsl_rpmsg_i2s.c
+++ b/sound/soc/fsl/fsl_rpmsg_i2s.c
@@ -200,7 +200,7 @@ static int fsl_rpmsg_i2s_suspend(struct device *dev)
 	rpmsg_tx->header.cmd = I2S_TX_SUSPEND;
 	i2s_send_message(rpmsg_tx, i2s_info);
 
-	rpmsg_tx->header.cmd = I2S_RX_SUSPEND;
+	rpmsg_rx->header.cmd = I2S_RX_SUSPEND;
 	i2s_send_message(rpmsg_rx, i2s_info);
 
 	return 0;
@@ -219,7 +219,7 @@ static int fsl_rpmsg_i2s_resume(struct device *dev)
 	rpmsg_tx->header.cmd = I2S_TX_RESUME;
 	i2s_send_message(rpmsg_tx, i2s_info);
 
-	rpmsg_tx->header.cmd = I2S_RX_RESUME;
+	rpmsg_rx->header.cmd = I2S_RX_RESUME;
 	i2s_send_message(rpmsg_rx, i2s_info);
 
 	return 0;
-- 
1.7.9.5

